docs/topics/power-assert.md

```diff
@@ -0,0 +1,399 @@
+
+class SampleTest {
+
+    @Test
+    fun testFunction() {
+        val hello = "Hello"
+        val world = "world!"
+        assert(hello.length == world.substring(1, 4).length) { "Incorrect length" }
+    }
+}
+```
+
+If you run the `testFunction()` test with the Power-assert plugin enabled, you get the explicit failure message:
+
+```text
+Incorrect length
+assert(hello.length == world.substring(1, 4).length) { "Incorrect length" }
+       |     |      |  |     |               |
+       |     |      |  |     |               3
+       |     |      |  |     orl
+       |     |      |  world!
+       |     |      false
+       |     5
+       Hello
+```
+
+To get a more complete error message, always inline the variable into the test function parameters.
+Consider the following test function:
+
+```kotlin
+class ComplexExampleTest {
+
+    data class Person(val name: String, val age: Int)
+ 
+    @Test
+    fun testComplexAssertion() {
+        val person = Person("Alice", 10)
+        val isValidName = person.name.startsWith("A") && person.name.length > 3
+        val isValidAge = person.age in 21..28
+        assert(isValidName && isValidAge)
+    }
+}
+```
+
+The output of the executed code doesn't provide enough information to find the cause of the problem:
+
+```text
+Assertion failed
+assert(isValidName && isValidAge)
+       |              |
+       |              false
+       true
+```
+
+Inline the variable into the `assert()` function:
+
+```kotlin
+class ComplexExampleTest {
+
+    data class Person(val name: String, val age: Int)
+
+    @Test
+    fun testComplexAssertion() {
+        val person = Person("Alice", 10)
+        assert(person.name.startsWith("A") && person.name.length > 3 && person.age > 20 && person.age < 29)
+    }
+}
+```
+
+After execution, you get more explicit information about what went wrong:
+
+```text
+Assertion failed
+assert(person.name.startsWith("A") && person.name.length > 3 && person.age > 20 && person.age < 29)
+       |      |    |                  |      |    |      |      |      |   |
+       |      |    |                  |      |    |      |      |      |   false
+       |      |    |                  |      |    |      |      |      10
+       |      |    |                  |      |    |      |      Person(name=Alice, age=10)
+       |      |    |                  |      |    |      true
+       |      |    |                  |      |    5
+       |      |    |                  |      Alice
+       |      |    |                  Person(name=Alice, age=10)
+       |      |    true
+       |      Alice
+       Person(name=Alice, age=10)
+```
+
+### Beyond assert function
+
+The Power-assert plugin can transform various functions beyond `assert` which is transformed by default.
+Functions like `require()`, `check()`, `assertTrue()`, `assertEqual()` and others can also be transformed,
+if they have a form that allows taking a `String` or `() -> String` value as the last parameter.
```

Comment:
Just as a note for a followup, I would like to see this expanded into more details in the future. It's enough for now, but we will probably get questions about the exact requirements, and more details here should help.

Reply:
Yeah, let's plan it as a next step! Thanks for you feedback!